Source/CBLSyncConnection+Pull.m

```diff
@@ -162,11 +164,17 @@ - (void) handleIncomingRevision: (BLIPRequest*)request {
     NSDictionary* attachments = nil;
     NSString* docID;
     NSData* json = request.body;
-    if (memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
+    if (self.onSyncAccessCheck && memmem(json.bytes, json.length, "\"_attachments\":", 15) != NULL) {
```

Comment:
Actually I think it would be better to remove the `docID` parameter from the access check block. It isn't actually used. The reason for this `if` test is to avoid having to parse the doc's JSON unless necessary, for speed, and this patch would cause it to always get parsed, for no good reason, if the db is read-only.